src/test/java/com/metamx/http/client/JankyServersTest.java

```diff
@@ -246,18 +257,29 @@ public void testHttpsConnectionClosingServer() throws Throwable
       }
       catch (ExecutionException e1) {
         e = e1.getCause();
+        e1.printStackTrace();
       }
 
-      Assert.assertTrue("ChannelException thrown by 'get'", e instanceof ChannelException);
+      Assert.assertTrue("ChannelException thrown by 'get'", isChannelClosedException(e));
     }
     finally {
       lifecycle.stop();
     }
+    System.out.println("Done testHttpsConnectionClosingServer");
+
+  }
+
+  public boolean isChannelClosedException(Throwable e)
+  {
+    return e instanceof ChannelException ||
+           (e instanceof IOException && e.getMessage().matches(".*Connection reset by peer.*"));
```

Comment:
On travis, When an exception is thrown , in some cases it causes the channel to be closed with an IOException